scripts/hostcfgd

```diff
@@ -204,12 +206,54 @@ class FeatureHandler(object):
     FEATURE_STATE_DISABLED = "disabled"
     FEATURE_STATE_FAILED = "failed"
 
-    def __init__(self, config_db, feature_state_table, device_config):
+    def enableDelayedServices(self):
+        self.lock.acquire()
+        self.enable_delayed_service = True
+        for feature_name in self._cached_config:
+            if self._cached_config[feature_name].delayed:
+                self.update_feature_state(self._cached_config[feature_name])
+        self.lock.release()
+
+    def portListener(self):
+        syslog.syslog(syslog.LOG_INFO, "Starting port listener")
+        appl_db_connector = DBConnector("APPL_DB", 0)
+        appl_db_id = SonicDBConfig.getDbId("APPL_DB")
+        subscribe_port = swsscommon.SubscriberStateTable(appl_db_connector, "PORT_TABLE")
+        sel = swsscommon.Select()
+        sel.addSelectable(subscribe_port)
+
+        if (self.is_advanced_boot):
+             syslog.syslog(syslog.LOG_INFO, "Updating delayed features after warm/fast boot")
+             self.enableDelayedServices()
+
+        while True:
+            (state, selectableObj) = sel.select(PORT_INIT_TIMEOUT_SEC*1000)
+            # Continue if select is timeout or selectable object is not return
+            if state != swsscommon.Select.OBJECT:
+                if not self.enable_delayed_service:
```

Comment:
If we were not asked to enable_delayed_service why did we start this thread at the first place? Or do I missunderstand the purpose of this flag?

Reply:
This logic is to enable delayed services when PortInitDone is not received even after 180 second timeout. We will check after timeout to see if delayed service enabled is set. If not we enable it.